operator/pkg/ingress/envoy_config.go

```diff
@@ -380,18 +388,43 @@ func getRouteMatch(ingressPath slim_networkingv1.HTTPIngressPath) *envoy_config_
 					Regex:      fmt.Sprintf("%s(/.*)?$", removedTrailingSlash),
 				},
 			},
+			Headers: headerMatchers,
 		}
 	}
 	if *ingressPath.PathType == slim_networkingv1.PathTypeExact {
 		return &envoy_config_route_v3.RouteMatch{
 			PathSpecifier: &envoy_config_route_v3.RouteMatch_Path{
 				Path: ingressPath.Path,
 			},
+			Headers: headerMatchers,
 		}
 	}
 	return nil
 }
 
+func getHeaderMatchers(host string) []*envoy_config_route_v3.HeaderMatcher {
+	if len(host) == 0 || host == wildCard || !strings.Contains(host, wildCard) {
+		return nil
+	}
+	// Make sure that wildcard character only match one single dns domain.
+	// For example, if host is *.foo.com, baz.bar.foo.com should not match
+	return []*envoy_config_route_v3.HeaderMatcher{
+		{
+			Name: envoyAuthority,
+			HeaderMatchSpecifier: &envoy_config_route_v3.HeaderMatcher_StringMatch{
+				StringMatch: &matcherv3.StringMatcher{
+					MatchPattern: &matcherv3.StringMatcher_SafeRegex{
+						SafeRegex: &matcherv3.RegexMatcher{
+							EngineType: &matcherv3.RegexMatcher_GoogleRe2{},
+							Regex:      strings.ReplaceAll(host, wildCard, noDotRegex),
```

Comment:
You can't characters that would be interpreted as regex non-literal matches (such as `.` (dot)) into a regex, as it will match any character. For example, the above will translate `*.google.com` to  regex `[^.]+.google.com` which will match `notgoogle.com`!

So you'll have to:
- replace all dots with `[.]` to enforce matching only a dot
- make sure there are no wildcards (`*`) anywhere else than in the beginning, and that it is followed by a `.`
- generally anchoring the regex is a good idea, unless you find documentation that Envoy will not match a substring (meaning that Envoy anchors the regex for you)
- Bonus point for adding a (unit) test case verifying that above, as it can be a security hazard if this is not done right.

Reply:
So the input string must be validated to only contain valid characters for a domain name (https://www.ietf.org/rfc/rfc1034.txt):
- [a-zA-Z0-9-]
- We've also had a case where users were using underscores (`_`) in domain names, so allow also those.
- replace potential `*.` in the beginning with `[^.]+[.]`
- replace each remaining `.` with `[.]`

Reply:
do we need to do the first and second validation, as host in ingress spec is following https://datatracker.ietf.org/doc/html/rfc3986, which also conforms with 1034 ?

Reply:
No need to validate character set if it is already validated.